tests/python/pants_test/engine/test_isolated_process.py

```diff
@@ -314,5 +376,5 @@ def mk_example_fs_tree(self):
     return fs_tree
 
   def mk_scheduler_in_example_fs(self, rules):
-    rules = list(rules) + create_fs_rules()
+    rules = list(rules) + create_fs_rules() + [RootRule(ExecuteProcessRequest)]
```

Comment:
Intrinsics are supposed to be ignored for the purposes of unused/unreachable rule checking... are you seeing otherwise?

Reply:
Yep, if I remove this, I see:

```
$ ./pants test tests/python/pants_test/engine:isolated_process -- -s -k test_write_file

10:45:42 00:00 [main]
               (To run a reporting server: ./pants server)
10:45:43 00:01   [setup]
10:45:49 00:07     [parse]
               Executing tasks in goals: bootstrap -> imports -> unpack-jars -> deferred-sources -> gen -> jvm-platform-validate -> resolve -> resources -> compile -> pyprep -> test
10:45:50 00:08   [bootstrap]
10:45:50 00:08     [substitute-aliased-targets]
10:45:50 00:08     [jar-dependency-management]
10:45:50 00:08     [bootstrap-jvm-tools]
10:45:50 00:08     [provide-tools-jar]
10:45:50 00:08   [imports]
10:45:50 00:08     [ivy-imports]
10:45:50 00:08   [unpack-jars]
10:45:50 00:08     [unpack-jars]
10:45:50 00:08   [deferred-sources]
10:45:50 00:08     [deferred-sources]
10:45:50 00:08   [gen]
10:45:50 00:08     [antlr-java]
10:45:50 00:08     [antlr-py]
10:45:50 00:08     [jaxb]
10:45:50 00:08     [protoc]
10:45:50 00:08     [ragel]
10:45:50 00:08     [thrift-java]
10:45:50 00:08     [thrift-py]
10:45:50 00:08     [wire]
10:45:50 00:08     [avro-java]
10:45:50 00:08     [go-thrift]
10:45:50 00:08     [jax-ws]
10:45:50 00:08     [scrooge]
10:45:50 00:08     [thrifty]
10:45:50 00:08   [jvm-platform-validate]
10:45:50 00:08     [jvm-platform-validate]
10:45:50 00:08   [resolve]
10:45:50 00:08     [ivy]
10:45:50 00:08     [coursier]
10:45:50 00:08     [go]
10:45:50 00:08     [scala-js-compile]
10:45:50 00:08     [scala-js-link]
10:45:50 00:08     [node]
10:45:50 00:08   [resources]
10:45:50 00:08     [prepare]
10:45:50 00:08     [services]
10:45:50 00:08   [compile]
10:45:50 00:08     [node]
10:45:50 00:08     [compile-jvm-prep-command]
10:45:50 00:08       [jvm_prep_command]
10:45:50 00:08     [compile-prep-command]
10:45:50 00:08     [compile]
10:45:50 00:08     [zinc]
10:45:50 00:08     [javac]
10:45:50 00:08     [cpp]
10:45:50 00:08     [errorprone]
10:45:50 00:08     [findbugs]
10:45:50 00:08     [go]
10:45:50 00:08   [pyprep]
10:45:50 00:08     [interpreter]
10:45:50 00:08     [build-local-dists]
10:45:50 00:08     [requirements]
10:45:50 00:08     [sources]
10:45:50 00:08   [test]
10:45:50 00:08     [test-jvm-prep-command]
10:45:50 00:08       [jvm_prep_command]
10:45:50 00:08     [test-prep-command]
10:45:50 00:08     [test]
10:45:50 00:08     [pytest-prep]
10:45:51 00:09     [pytest]
                   Invalidated 2 targets.
                   scrubbed PYTHONPATH=/Users/dwagnerhall/src/github.com/pantsbuild/otherpants/src/python: from py.test environment
10:45:51 00:09       [run]
                     ============== test session starts ===============
                     platform darwin -- Python 2.7.14, pytest-3.4.2, py-1.5.2, pluggy-0.6.0
                     rootdir: /Users/dwagnerhall/src/github.com/pantsbuild/otherpants/.pants.d, inifile: /Users/dwagnerhall/src/github.com/pantsbuild/otherpants/.pants.d/test/pytest-prep/CPython-2.7.14/5a577850dda9b793dbf4d39d014db37ec4327bca/pytest.ini
                     plugins: timeout-1.2.1, cov-2.4.0
                     collected 6 items
                     
                     tests/python/pants_test/engine/test_isolated_process.py F
                     
                      generated xml file: /Users/dwagnerhall/src/github.com/pantsbuild/otherpants/.pants.d/test/pytest/914ece5fb25f3ae64b1832fd03888b46361db439/junitxml/TEST-914ece5fb25f3ae64b1832fd03888b46361db439.xml 
                     ==================== FAILURES ====================
                     ______ IsolatedProcessTest.test_write_file _______
                     
                     self = <pants_test.engine.test_isolated_process.IsolatedProcessTest testMethod=test_write_file>
                     
                         def test_write_file(self):
                           scheduler = self.mk_scheduler_in_example_fs(())
                         
                           request = ExecuteProcessRequest.create_with_empty_snapshot(
                             ("/bin/bash", "-c", "echo -n 'European Burmese' > roland"),
                             dict(),
                             ("roland",)
                           )
                         
                     >     execute_process_result = self.execute_expecting_one_result(scheduler, ExecuteProcessResult, request).value
                     
                     .pants.d/pyprep/sources/25b99150a0661ed71e060bf1d326fcd9be91c377/pants_test/engine/test_isolated_process.py:303: 
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     .pants.d/pyprep/sources/25b99150a0661ed71e060bf1d326fcd9be91c377/pants_test/engine/scheduler_test_base.py:79: in execute_expecting_one_result
                         request = scheduler.execution_request([product], [subject])
                     .pants.d/pyprep/sources/25b99150a0661ed71e060bf1d326fcd9be91c377/pants/engine/scheduler.py:374: in execution_request
                         self._scheduler.add_root_selection(native_execution_request, subject, product)
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     
                     self = <pants.engine.scheduler.Scheduler object at 0x10fdb5150>
                     execution_request = <cdata 'void *' 0x7fd94ca8a870>
                     subject = ExecuteProcessRequest(argv=(u'/bin/bash', u'-c', u"echo -n 'European Burmese' ...st(fingerprint=e3b0c442, serialized_bytes_length=0), output_files=(u'roland',))
                     product = <class 'pants.engine.isolated_process.ExecuteProcessResult'>
                     
                         def add_root_selection(self, execution_request, subject, product):
                           res = self._native.lib.execution_add_root_select(self._scheduler,
                                                                            execution_request,
                                                                            self._to_key(subject),
                                                                            self._to_constraint(product))
                           if res.is_throw:
                     >       raise self._from_value(res.value)
                     E       Exception: No installed rules can satisfy Select(ExecuteProcessResult) for a root subject of type ExecuteProcessRequest.
                     
                     .pants.d/pyprep/sources/25b99150a0661ed71e060bf1d326fcd9be91c377/pants/engine/scheduler.py:283: Exception
                     =============== 5 tests deselected ===============
                     ===== 1 failed, 5 deselected in 0.60 seconds =====
                     
                   tests/python/pants_test/engine:isolated_process                                 .....   FAILURE
                   tests/python/pants_test/engine:scheduler_test_base                              .....   SUCCESS
FAILURE: FAILURE


               Waiting for background workers to finish.
10:45:52 00:10   [complete]
               FAILURE
```

Leaving this in for now, and we can follow up